app/controllers/state_file/questions/data_review_controller.rb

```diff
@@ -1,12 +1,27 @@
 module StateFile
   module Questions
     class DataReviewController < QuestionsController
-      private
+      def edit
+        super
+        StateFileEfileDeviceInfo.find_or_create_by!(
+          event_type: "initial_creation",
+          ip_address: ip_for_irs,
+          intake: current_intake,
+        )
+      end
 
-      def form_class
-        NullForm
+      def update
+        if form_params["device_id"].blank?
+          flash[:alert] = I18n.t("general.enable_javascript")
```

Comment:
if the client does not have javascript enabled we won't be able to grab the device id; in CTC we also displayed this "please enable javascript message" if there was no javascript. Although might be worth checking if we want to be so harsh as to not let them continue without a device id. We do need it inorder for the XML to pass but to what end will IRS need one to accept the return.

Reply:
Just to note, asked Alisa on the DF team and they said they would ask their team what they are doing